src/Build/BackEnd/BuildManager/BuildParameters.cs

```diff
@@ -472,6 +479,11 @@ public IDictionary<string, string> GlobalProperties
             }
         }
 
+        /// <summary>
+        /// Properties passed from the command line (e.g. by using /p:).
+        /// </summary>
+        public HashSet<string> PropertiesFromCommandLine => _propertiesFromCommandLine;
```

Comment:
Ah, I think this isn't quite what we want, instead we want "global properties set for the whole build", independent of whether that's through the API or command line args.

Reply:
I think we already have it, don't we ?
https://github.com/dotnet/msbuild/blob/0d310deb955150d91a86b43af9e0cd1690d51690/src/Build/Evaluation/PropertyTrackingEvaluatorDataWrapper.cs#L90C78-L90C94

I was adding it to fulfill this requirement
https://github.com/dotnet/msbuild/issues/2711#:~:text=have%20it%20already)%3A-,property,-coming%20from%20a

Reply:
could you please clarify if we need this additional classification or not?

Reply:
We shouldn't expose HashSet<T> in public API. If we're adding a new public API here, what are the usages? Could it be just an ICollection<T>? Where is this used? Does this need to be new public API?